deploy/wmagent-venv-deploy.sh

```diff
@@ -0,0 +1,760 @@
+        command -v $tool 2>&1 > /dev/null || {
+            error=$?;
+            echo "The current setup script requires: $tool in order to continue. Please install it and rerun." ;
+            return $error ;}
+    done
+}
+
+_sort(){
+    # Simple auxiliary sort function.
+    # :param $*: All parameters need to be string values to be sorted
+    # :return:   Prints the alphabetically sorted list of all input parameters
+    local -a result
+    i=0
+    result[$i]=$1
+    shift
+    for key in $*
+    do
+        let i++
+        x=$i
+        y=$i
+        result[$i]=$key
+        while [[ $x -gt 0 ]]
+        do
+            let x--
+            if [[ ${result[$x]} > ${result[$y]} ]]; then
+                tmpKey=${result[$x]}
+                result[$x]=${result[$y]}
+                result[$y]=$tmpKey
+            else
+                break
+            fi
+            let y--
+        done
+    done
+    echo ${result[*]}
+}
+
+printVenvSetup(){
+    # Function to print the current virtual environment setup. And a basic
+    # deployment area tree, no deeper than 3 levels relative to the deployment
+    # root path.
+    # :param:  None
+    # :return: Dumps a formatted string with the information for the current setup
+
+    echo "======================================================="
+    echo "Printing the final WMCore virtual environment parameters and tree:"
+    echo "-------------------------------------------------------"
+    echo
+    tree -d -L 3 $venvPath
+    echo
+    echo "-------------------------------------------------------"
+
+    local prefix="WMCoreVenvVars[]: "
+    local prefixLen=${#prefix}
+
+    # NOTE: Here choosing the common alignment position for all variables printed
+    #       the position count starts from the beginning of line + prefixLen.
+    local valAllign=0
+    for var in ${!WMCoreVenvVars[@]}
+    do
+        vLen=${#var}
+        [[ $valAllign -lt $vLen ]] && valAllign=$vLen
+    done
+
+    for var in $(_sort ${!WMCoreVenvVars[@]})
+    do
+        vLen=${#var}
+        spaceLen=$(($valAllign - $vLen))
+        space=""
+        for ((i=0; i<=$spaceLen; i++))
+        do
+            space="$space "
+        done
+        spaceNewLineLen=$(($spaceLen + $prefixLen +$vLen))
+        spaceNewLine=""
+        for ((i=0; i<=$spaceNewLineLen; i++))
+        do
+            spaceNewLine="$spaceNewLine "
+        done
+        echo -e "WMCoreVenvVars[$var]${space}: ${WMCoreVenvVars[$var]//:/\n$spaceNewLine}"
+
+    done
+}
+
+wmaInstall() {
+    # The main function to setup/add the WMAgent virtual environment variables
+    # and call the install.sh script from: https://github.com/dmwm/CMSKubernetes/blob/master/docker/pypi/wmagent/install.sh
+
+    venvPath=$(_realPath $venvPath)
+    # TODO: To be changed to point to master at https://github.com/dmwm/CMSKubernetes.git
+    #       once we have it all merged at CMSKubernetes
+    deployRepo=https://github.com/todor-ivanov/CMSKubernetes.git
```

Comment:
Todor, for the future, please make a clear note (TODO) in the PR description that such lines need to be corrected before merging.

Reply:
The PR description has been updated.

Reply:
Should this be updated now?

Reply:
done